PowerValidatedSolutions.psm1

```diff
@@ -48145,6 +48145,383 @@ Function Test-WMSubnetInput {
+
+        .EXAMPLE
+        Invoke-RequestSignedCertificateByMsca -caFqdn "rpl-ad01.rainpole.io" -username "Administrator" -password "VMw@re1!" -csrFilePath
+         "C:\certificates\xint-vrli01.rainpole.io.csr" -certDir "C:\certificates" -certificateTemplate "VMware" -getCArootCert
+        This example will request the xint-vrli01 CSR file to be signed by Certificate Authority rpl-ad01.rainpole.io and also will retrieve 
+        the Certificate Authority's root chain certificate.  
+
+        .PARAMETER caFqdn
+        The FQDN of the Certificate Authorty web enrollment website.
+
+        .PARAMETER username
+        The username to authenticate to the Certificate Authorty web enrollment website.
+
+        .PARAMETER password
+        The password to authenticate to the Certificate Authorty web enrollment website.
+
+        .PARAMETER csrFilePath
+        The full file path for the certificate signing request file. 
+
+        .PARAMETER certDir
+        The directory path to store the certificate authority signed certificate file.
+
+        .PARAMETER certificateTemplate
+        The name of the certificate template to be used.  
+
+        .PARAMETER getCArootCert
+        The Certificate Authorty chain root certificate will be retrieved as well.   
+    #>
+
+    Param (
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $caFqdn,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $username,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $password,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $csrFilePath,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $certDir,            
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $certificateTemplate,
+        [Parameter (Mandatory = $false)] [Switch] $getCArootCert
+    )
+
+    # Validate inputs
+    if (-Not (Test-Path -Path $certDir)) {
+        Write-Error "Error: $certDir directory does not exist." -ErrorAction Stop
+    }
+    if (-Not (Test-Path -Path $csrFilePath)) {
+        Write-Error "Error: $csrFilePath file not found." -ErrorAction Stop
+    }
+    $fileName = Split-Path -Path $csrFilePath -LeafBase
+    $crtFileName = $fileName + ".crt"
+    $crtFilePath = Join-Path -Path $certDir -ChildPath $crtFileName
+    if (Test-Path -Path $crtFilePath) {
+        Write-Error "Error: $crtFilePath file exists.  Please remove or rename the existing file." -ErrorAction Stop
+    }
+
+    $securePass = ConvertTo-SecureString -String $password -AsPlainText -Force
+    $domainCreds = New-Object System.Management.Automation.PSCredential ($username, $securePass)
+    
+    Try {
+        $respond = Invoke-WebRequest -Uri "https://$caFqdn/certsrv/certrqxt.asp" -Credential $domainCreds -Method Get -SkipCertificateCheck
+    } catch {
```

Comment:
```suggestion
    } Catch {
```

Reply:
Addressed in commit 53039acbaaffafaf219274cac1a3684735130f53